org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java

```diff
@@ -100,38 +106,51 @@ private void importDir(Path rootFolder, IProgressMonitor monitor) {
 				}
 			}
 		}
-		startSynchronization(rootFolder.toFile(), distribution, NewProjectHandler.IMPORT_AND_MERGE);
+		return distribution;
 	}
 
-	protected void startSynchronization(File location, GradleDistribution distribution, NewProjectHandler newProjectHandler) {
-		List<String> jvmArgs = new ArrayList<>();
-		IProxyService proxyService = JavaLanguageServerPlugin.getInstance().getProxyService();
-		IProxyData proxy = proxyService.getProxyData(IProxyData.HTTP_PROXY_TYPE);
-		if (proxy != null) {
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTP_PROXY_HOST, proxy.getHost());
-			if (proxy.getPort() > 0) {
-				addArgs(jvmArgs, JavaLanguageServerPlugin.HTTP_PROXY_PORT, String.valueOf(proxy.getPort()));
+	protected void startSynchronization(Path rootFolder, IProgressMonitor monitor) {
+		boolean shouldSynchronize = true;
+		File location = rootFolder.toFile();
+		List<IProject> projects = ProjectUtils.getGradleProjects();
+		for (IProject project : projects) {
+			File projectDir = project.getLocation() == null ? null : project.getLocation().toFile();
+			if (location.equals(projectDir)) {
+				shouldSynchronize = checkGradlePersistence(shouldSynchronize, project, projectDir);
+				break;
 			}
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTP_PROXY_USER, proxy.getUserId());
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTP_PROXY_PASSWORD, proxy.getPassword());
 		}
-		proxy = proxyService.getProxyData(IProxyData.HTTPS_PROXY_TYPE);
-		if (proxy != null) {
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTPS_PROXY_HOST, proxy.getHost());
-			if (proxy.getPort() > 0) {
-				addArgs(jvmArgs, JavaLanguageServerPlugin.HTTPS_PROXY_PORT, String.valueOf(proxy.getPort()));
-			}
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTPS_PROXY_USER, proxy.getUserId());
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTPS_PROXY_PASSWORD, proxy.getPassword());
+		if (shouldSynchronize) {
+			GradleDistribution distribution = getGradleDistribution(rootFolder);
+			BuildConfiguration configuration = CorePlugin.configurationManager().createBuildConfiguration(location, false, distribution, null, false, false, false);
+			GradleBuild build = CorePlugin.gradleWorkspaceManager().getGradleBuild(configuration);
+			build.getModelProvider().fetchEclipseGradleProjects(FetchStrategy.LOAD_IF_NOT_CACHED, GradleConnector.newCancellationTokenSource().token(), monitor);
+			build.synchronize(NewProjectHandler.IMPORT_AND_MERGE);
 		}
-		FixedRequestAttributes attributes = new FixedRequestAttributes(location, null, distribution, null, jvmArgs, Collections.emptyList());
-		CorePlugin.gradleWorkspaceManager().getGradleBuild(attributes).synchronize(newProjectHandler);
 	}
 
-	private void addArgs(List<String> jvmArgs, String name, String value) {
-		if (StringUtils.isNotBlank(value)) {
-			jvmArgs.add(String.format("-D%s=%s", name, value));
+	private boolean checkGradlePersistence(boolean shouldSynchronize, IProject project, File projectDir) {
```

Comment:
can you please add some ITs for that piece of logic?

Reply:
See https://github.com/eclipse/eclipse.jdt.ls/pull/459#issuecomment-346693688